utils/modCommon.js

```diff
@@ -678,26 +678,24 @@ const modCommon = {
    * Give somebody a staff assigned role
    * @param {Augur.GuildInteraction<"CommandSlash">} int
    * @param {Boolean} give
-   * @param {Discord.GuildMember | null} recipient
+   * @param {Discord.GuildMember} recipient
+   * @param {string} input
    * @returns {Promise<string>}
    */
-  assignRole: async function(int, recipient, give = true) {
+  assignRole: async function(int, recipient, input, give = true) {
     /** @param {Discord.GuildMember} member @param {string} id*/
-    const hasRole = (member, id) => member?.roles.cache.has(id);
     try {
       const pres = give ? "give" : "take";
       const past = give ? "gave" : "took";
-      const input = int.options.getString("role", true);
       if (!u.perms.calc(int.member, ["team", "mod", "mgr"])) return "*Nice try!* This command is for Team+ only";
       if (!u.perms.calc(int.member, ["mod", "mgr"]) && ["adulting", "lady"].includes(input.toLowerCase())) return "This command is for Mod+ only";
-      if (!recipient) return `I couldn't find that user!`;
       const role = int.guild.roles.cache.find(r => r.name.toLowerCase() == input.toLowerCase());
```

Comment:
The role should be found in the main function. This should just add the role and check for permissions.